src/frequenz/sdk/actor/power_distributing/_battery_status_utils.py

```diff
@@ -0,0 +1,131 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+"""Helper classes used only by BatteryStatus."""
+
+import datetime
+from dataclasses import dataclass, field
+from typing import Dict
+
+from frequenz.channels import Peekable
+
+from ...microgrid.component import BatteryData, InverterData
+
+
+@dataclass(frozen=True)
+class BatInvReceivers:
+    """Store receivers for battery and the adjacent inverter."""
+
+    # Receiver for given battery
+    bat_recv: Peekable[BatteryData]
+    # Receiver for an inverter adjacent to that battery
+    inv_recv: Peekable[InverterData]
+
+
+@dataclass
+class _TimeoutParam:
+    """Set timeout for component, and track when it is finished."""
+
+    start_time: datetime.datetime
+    max_timeout_duration_s: float
+    duration_s: float = 2
+    expiration_time: datetime.datetime = field(init=False)
+
+    def __post_init__(self) -> None:
+        """Create parameters based on the input."""
+        self.expiration_time = self.start_time + datetime.timedelta(
+            seconds=self.duration_s
+        )
+
+    def increase_timeout(self, now: datetime.datetime) -> float:
+        """Update timeout.
+
+        Next timeout will be twice longer then the previous one.
+        Maximum duration time for the timeout is self.max_timeout_duration_s.
+
+        Args:
+            now: current time
+
+        Returns:
+            Return for how long the timeout has been set. Return 0 if the previous
+                timeout is till not finished and new timeout was not set.
+        """
+        if self.expiration_time <= now:
+            self.duration_s = min(self.duration_s * 2, self.max_timeout_duration_s)
+            self.expiration_time = now + datetime.timedelta(seconds=self.duration_s)
+            return self.duration_s
+        return 0
+
+    def is_expired(self, now: datetime.datetime) -> bool:
+        """Return True if timer expired.
+
+        Args:
+            now: current time
+
+        Returns:
+            True if timer expired, false otherwise.
+        """
+        return self.expiration_time <= now
+
+
+class BlockedComponents:
```

Comment:
Maybe it makes sense to add this to component-specific classes, instead of being separate?

Then you could run functions like
```
if battery_0.is_blocked():
    bla()
```

You could also add a method `blocked_until()`, which returns the timestamp until when the component is blocked.

Reply:
Done